gulpfile.js

```diff
@@ -0,0 +1,105 @@
+
+var gulp = require('gulp'),
+  runSequence = require('run-sequence'),
+  del = require('del'),
+  replace = require('gulp-string-replace'),
+  sourcemaps = require('gulp-sourcemaps'),
+  exec = require('child_process').exec,
+  ngc = require('gulp-ngc'),
+  changed = require('gulp-changed');
+
+var appSrc = 'src';
+var libraryDist = 'dist';
+var watchDist = 'dist-watch';
+var globalExcludes = [ '!./**/examples/**', '!./**/examples' ]
+
+/**
+ * FUNCTION LIBRARY
+ */
+
+function copyToDist(srcArr) {
+  return gulp.src(srcArr.concat(globalExcludes))
+    .pipe(gulp.dest(function (file) {
+      return libraryDist + file.base.slice(__dirname.length); // save directly to dist
+    }));
+}
+
+function updateWatchDist() {
+  return gulp
+    .src([libraryDist + '/**'].concat(globalExcludes))
+    .pipe(changed(watchDist))
+    .pipe(gulp.dest(watchDist));
+}
+
+/**
+ * TASKS
+ */
+
+gulp.task('post-transpile', ['transpile'], function () {
+  return gulp.src(['dist/src/app/**/*.js'])
+    .pipe(replace(/templateUrl:\s/g, "template: require("))
+    .pipe(replace(/\.html',/g, ".html'),"))
+    .pipe(replace(/\.html'/g, ".html')"))
+    .pipe(replace(/styleUrls: \[/g, "styles: [require("))
+    .pipe(replace(/\.scss']/g, ".css').toString()]"))
+    .pipe(gulp.dest(function (file) {
+      return file.base; // because of Angular 2's encapsulation, it's natural to save the css where the scss-file was
+    }));
+});
+
+//Sass compilation and minifiction
+gulp.task('transpile-sass', function () {
+  return transpileSASS(appSrc + '/app/**/*.scss');
+});
+
+// Put the SASS files back to normal
+gulp.task('build-library',
+  [
+    'transpile',
+    'post-transpile',
```

Comment:
I think that much of this is wasted on this library as it is only services. You should keep the gulp commands but only transpile the .ts. 

Currently there is no .html and no .scss.